src/sinks/loki/sink.rs

```diff
@@ -238,6 +238,46 @@ impl EventEncoder {
     }
 }
 
+enum FilteredRecord {
+    Rewritten(LokiRecord),
+    Valid(LokiRecord),
+}
```

Comment:
Since both variants here wrap the same data, I'm not seeing that this adds much value compared to the complexity of dealing with all the `inner` bits, compared to a tuple of a simple enum  and the value.